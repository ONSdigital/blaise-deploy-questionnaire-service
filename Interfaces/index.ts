interface Instrument {
    installDate: string
    name: string
    expired: boolean
    serverParkName: string
    activeToday: boolean
    surveyDays: string[]
    link: string
    fieldPeriod: string
    surveyTLA: string
}

interface Survey {
    instruments: Instrument[]
    survey: string
}

export type {Instrument, Survey};
