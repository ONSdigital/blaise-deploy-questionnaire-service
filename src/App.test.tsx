import React from "react";
import Enzyme from "enzyme";
import {render, waitFor, fireEvent, screen, cleanup} from "@testing-library/react";
import Adapter from "enzyme-adapter-react-16";
import App from "./App";
import "@testing-library/jest-dom";
import flushPromises from "./tests/utils";
import {act} from "react-dom/test-utils";
import {createMemoryHistory} from "history";
import {Router} from "react-router";
import {Survey} from "../Interfaces";

const surveyListReturned: Survey[] = [
    {
        survey: "OPN",
        instruments: [
            {
                activeToday: true,
                fieldPeriod: "July 2020",
                expired: false,
                installDate: "2020-12-11T11:53:55.5612856+00:00",
                link: "https://external-web-url/OPN2007T?LayoutSet=CATI-Interviewer_Large",
                name: "OPN2007T",
                serverParkName: "LocalDevelopment",
                "surveyTLA": "OPN",
                surveyDays: []
            }
        ]
    }
];

function mock_server_request(returnedStatus: number, returnedJSON: any) {
    global.fetch = jest.fn(() =>
        Promise.resolve({
            status: returnedStatus,
            json: () => Promise.resolve(returnedJSON),
        })
    );
}

describe("React homepage", () => {
    Enzyme.configure({adapter: new Adapter()});

    beforeAll(() => {
        mock_server_request(200, surveyListReturned);
    });


    it("view surveys page matches Snapshot", async () => {
        const history = createMemoryHistory();
        const wrapper = render(
            <Router history={history}>
                <App/>
            </Router>
        );

        await act(async () => {
            await flushPromises();
        });

        await waitFor(() => {
            expect(wrapper).toMatchSnapshot();
        });
    });

    it("view questionnaires page matches Snapshot", async () => {
        const history = createMemoryHistory();
        const wrapper = render(
            <Router history={history}>
                <App/>
            </Router>
        );

        await act(async () => {
            await flushPromises();
        });

        await fireEvent.click(screen.getByText(/View active questionnaires/i));

        await act(async () => {
            await flushPromises();
        });

        await waitFor(() => {
            expect(wrapper).toMatchSnapshot();
        });

    });

    it("should render correctly", async () => {
        const history = createMemoryHistory();
        const {getByText, queryByText } = render(
            <Router history={history}>
                <App/>
            </Router>
        );

        expect(queryByText(/Loading/i)).toBeInTheDocument();

        await waitFor(() => {
            expect(getByText(/Telephone Operations Blaise Interface/i)).toBeDefined();
            expect(getByText(/OPN/i)).toBeDefined();
            expect(queryByText(/Loading/i)).not.toBeInTheDocument();
        });

        await fireEvent.click(getByText(/View active questionnaires/i));

        await act(async () => {
            await flushPromises();
        });

        await waitFor(() => {
            expect(getByText(/Telephone Operations Blaise Interface/i)).toBeDefined();
            expect(getByText(/OPN2007T/i)).toBeDefined();
            expect(queryByText(/Loading/i)).not.toBeInTheDocument();
        });

    });

    afterAll(() => {
        jest.clearAllMocks();
        cleanup();
    });
});


describe("Given the API returns malformed json", () => {
    Enzyme.configure({adapter: new Adapter()});

    beforeAll(() => {
        mock_server_request(200, {text: "Hello"});
    });

    it("it should render with the error message displayed", async () => {
        const history = createMemoryHistory();
        const {getByText, queryByText } = render(
            <Router history={history}>
                <App/>
            </Router>
        );

        expect(queryByText(/Loading/i)).toBeInTheDocument();


        await waitFor(() => {
            expect(getByText(/Sorry, there is a problem with this service. We are working to fix the problem. Please try again later./i)).toBeDefined();
            expect(queryByText(/Loading/i)).not.toBeInTheDocument();
        });

    });

    afterAll(() => {
        jest.clearAllMocks();
        cleanup();
    });
});

describe("Given the API returns an empty list", () => {
    Enzyme.configure({adapter: new Adapter()});

    beforeAll(() => {
        mock_server_request(200, []);
    });

    it("it should render with a message to inform the user in the list", async () => {
        const history = createMemoryHistory();
        const {getByText, queryByText } = render(
            <Router history={history}>
                <App/>
            </Router>
        );

        expect(queryByText(/Loading/i)).toBeInTheDocument();


        await waitFor(() => {
            expect(getByText(/No active surveys found./i)).toBeDefined();
            expect(queryByText(/Loading/i)).not.toBeInTheDocument();
        });

    });

    afterAll(() => {
        jest.clearAllMocks();
        cleanup();
    });
});
