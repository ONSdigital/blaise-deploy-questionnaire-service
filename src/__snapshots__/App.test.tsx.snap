// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React homepage view questionnaires page matches Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="phase-banner"
      >
        <div
          class="container "
        >
          <div
            class="grid grid--flex grid--gutterless grid--vertical-center grid--no-wrap"
          >
            <div
              class="grid__col col-auto u-flex-no-grow"
            >
              <h3
                class="phase-banner__badge"
              >
                BETA
              </h3>
            </div>
            <div
              class="grid__col col-auto u-flex-shrink"
            >
              <p
                class="phase-banner__desc u-fs-s u-mb-no"
              >
                This is a new service – your 
                <a
                  href="https://ons.service-now.com/"
                >
                  feedback
                </a>
                 will help us improve it.
              </p>
            </div>
          </div>
        </div>
      </div>
      <header
        class="header header--internal"
      >
        <div
          class="header__top"
          role="banner"
        >
          <div
            class="container"
          >
            <div
              class="header__grid-top grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap "
            >
              <div
                class="grid__col col-auto"
              >
                <a
                  class="header__logo-link"
                  href="/"
                >
                  <picture>
                    <img
                      alt="Office for National Statistics logo"
                      class="header__logo"
                      src="https://ons-design-system.netlify.app/img/ons-logo-neg-en.svg"
                    />
                  </picture>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="header__main"
        >
          <div
            class="container"
          >
            <div
              class="grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap"
            >
              <div
                class="grid__col col-auto u-flex-shrink"
              >
                <div
                  class="header__title"
                >
                  Telephone Operations Blaise Interface
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
      <div
        class="page__container container"
        style="min-height: calc(67vh);"
      >
        <main
          class="page__main"
          id="main-content"
        >
          <h1>
            Interviewing
          </h1>
          <p>
            This page provides information on active questionnaires with corresponding links that redirect to specific areas of CATI dashboard.
          </p>
          <p>
            Please note, the table containing information on active questionnaires may take a few seconds to load.
          </p>
          <p
            class="u-mt-m"
          >
            <a
              class="external-link"
              id="cati-dashboard"
              rel="noopener noreferrer"
              target="_blank"
            >
              Link to CATI dashboard
              <svg
                class="svg-icon"
                viewBox="0 0 12 12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.5,9H13a.5.5,0,0,0-.5.5v3h-9v-9h3A.5.5,0,0,0,7,3V2.5A.5.5,0,0,0,6.5,2h-4a.5.5,0,0,0-.5.5v11a.5.5,0,0,0,.5.5h11a.5.5,0,0,0,.5-.5v-4A.5.5,0,0,0,13.5,9Z"
                  transform="translate(-2 -1.99)"
                />
                <path
                  d="M8.83,7.88a.51.51,0,0,0,.71,0l2.31-2.32,1.28,1.28A.51.51,0,0,0,14,6.49v-4a.52.52,0,0,0-.5-.5h-4A.51.51,0,0,0,9,2.52a.58.58,0,0,0,.14.33l1.28,1.28L8.12,6.46a.51.51,0,0,0,0,.71Z"
                  transform="translate(-2 -1.99)"
                />
              </svg>
            </a>
          </p>
          <p>
            <a
              href="/"
              id="return-to-survey-list"
            >
              Return to survey list
            </a>
          </p>
          <h2>
            Active questionnaires
          </h2>
          <table
            class="table "
            id="instrument-table"
          >
            <thead
              class="table__head u-mt-m"
            >
              <tr
                class="table__row"
              >
                <th
                  class="table__header "
                  scope="col"
                >
                  <span>
                    Questionnaire
                  </span>
                </th>
                <th
                  class="table__header "
                  scope="col"
                >
                  <span>
                    Field period
                  </span>
                </th>
                <th
                  class="table__header "
                  scope="col"
                >
                  <span>
                    Link to interview
                  </span>
                </th>
              </tr>
            </thead>
            <tbody
              class="table__body"
            >
              <tr
                class="table__row"
                data-testid="instrument-table-row"
              >
                <td
                  class="table__cell "
                >
                  OPN2007T
                </td>
                <td
                  class="table__cell "
                >
                  July 2020
                </td>
                <td
                  class="table__cell "
                >
                  <a
                    aria-label="Launch interview for instrument OPN2007T July 2020"
                    class="external-link"
                    href="https://external-web-url/OPN2007T?LayoutSet=CATI-Interviewer_Large"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Interview
                    <svg
                      class="svg-icon"
                      viewBox="0 0 12 12"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13.5,9H13a.5.5,0,0,0-.5.5v3h-9v-9h3A.5.5,0,0,0,7,3V2.5A.5.5,0,0,0,6.5,2h-4a.5.5,0,0,0-.5.5v11a.5.5,0,0,0,.5.5h11a.5.5,0,0,0,.5-.5v-4A.5.5,0,0,0,13.5,9Z"
                        transform="translate(-2 -1.99)"
                      />
                      <path
                        d="M8.83,7.88a.51.51,0,0,0,.71,0l2.31-2.32,1.28,1.28A.51.51,0,0,0,14,6.49v-4a.52.52,0,0,0-.5-.5h-4A.51.51,0,0,0,9,2.52a.58.58,0,0,0,.14.33l1.28,1.28L8.12,6.46a.51.51,0,0,0,0,.71Z"
                        transform="translate(-2 -1.99)"
                      />
                    </svg>
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
        </main>
      </div>
      <footer
        class="footer"
      >
        <div
          class="footer--body"
          data-analytics="footer"
        >
          <div
            class="container "
          >
            <div
              class="grid"
            >
              <div
                class="u-mb-m"
              >
                <dl
                  aria-label="Blaise Server Manager link to telephone server park"
                  class="metadata metadata__list grid grid--gutterless u-cf u-mb-l"
                  title="Blaise Server Manager link to telephone server park"
                >
                  <dt
                    class="metadata__term grid__col col-6@m"
                  >
                     Telephone server park for Blaise Server Manager:
                  </dt>
                  <dd
                    class="metadata__value grid__col col-6@m"
                  />
                </dl>
                <hr
                  class="footer__hr"
                />
              </div>
            </div>
            <div
              class="grid grid--flex grid--between"
            >
              <div
                class="grid__col"
              >
                <div
                  class="u-mt-m u-mb-s"
                >
                  <img
                    alt="Office for National Statistics"
                    src="https://ons-design-system.netlify.app/img/ons-logo-black-en.svg"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <div
      class="phase-banner"
    >
      <div
        class="container "
      >
        <div
          class="grid grid--flex grid--gutterless grid--vertical-center grid--no-wrap"
        >
          <div
            class="grid__col col-auto u-flex-no-grow"
          >
            <h3
              class="phase-banner__badge"
            >
              BETA
            </h3>
          </div>
          <div
            class="grid__col col-auto u-flex-shrink"
          >
            <p
              class="phase-banner__desc u-fs-s u-mb-no"
            >
              This is a new service – your 
              <a
                href="https://ons.service-now.com/"
              >
                feedback
              </a>
               will help us improve it.
            </p>
          </div>
        </div>
      </div>
    </div>
    <header
      class="header header--internal"
    >
      <div
        class="header__top"
        role="banner"
      >
        <div
          class="container"
        >
          <div
            class="header__grid-top grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap "
          >
            <div
              class="grid__col col-auto"
            >
              <a
                class="header__logo-link"
                href="/"
              >
                <picture>
                  <img
                    alt="Office for National Statistics logo"
                    class="header__logo"
                    src="https://ons-design-system.netlify.app/img/ons-logo-neg-en.svg"
                  />
                </picture>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="header__main"
      >
        <div
          class="container"
        >
          <div
            class="grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap"
          >
            <div
              class="grid__col col-auto u-flex-shrink"
            >
              <div
                class="header__title"
              >
                Telephone Operations Blaise Interface
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
    <div
      class="page__container container"
      style="min-height: calc(67vh);"
    >
      <main
        class="page__main"
        id="main-content"
      >
        <h1>
          Interviewing
        </h1>
        <p>
          This page provides information on active questionnaires with corresponding links that redirect to specific areas of CATI dashboard.
        </p>
        <p>
          Please note, the table containing information on active questionnaires may take a few seconds to load.
        </p>
        <p
          class="u-mt-m"
        >
          <a
            class="external-link"
            id="cati-dashboard"
            rel="noopener noreferrer"
            target="_blank"
          >
            Link to CATI dashboard
            <svg
              class="svg-icon"
              viewBox="0 0 12 12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.5,9H13a.5.5,0,0,0-.5.5v3h-9v-9h3A.5.5,0,0,0,7,3V2.5A.5.5,0,0,0,6.5,2h-4a.5.5,0,0,0-.5.5v11a.5.5,0,0,0,.5.5h11a.5.5,0,0,0,.5-.5v-4A.5.5,0,0,0,13.5,9Z"
                transform="translate(-2 -1.99)"
              />
              <path
                d="M8.83,7.88a.51.51,0,0,0,.71,0l2.31-2.32,1.28,1.28A.51.51,0,0,0,14,6.49v-4a.52.52,0,0,0-.5-.5h-4A.51.51,0,0,0,9,2.52a.58.58,0,0,0,.14.33l1.28,1.28L8.12,6.46a.51.51,0,0,0,0,.71Z"
                transform="translate(-2 -1.99)"
              />
            </svg>
          </a>
        </p>
        <p>
          <a
            href="/"
            id="return-to-survey-list"
          >
            Return to survey list
          </a>
        </p>
        <h2>
          Active questionnaires
        </h2>
        <table
          class="table "
          id="instrument-table"
        >
          <thead
            class="table__head u-mt-m"
          >
            <tr
              class="table__row"
            >
              <th
                class="table__header "
                scope="col"
              >
                <span>
                  Questionnaire
                </span>
              </th>
              <th
                class="table__header "
                scope="col"
              >
                <span>
                  Field period
                </span>
              </th>
              <th
                class="table__header "
                scope="col"
              >
                <span>
                  Link to interview
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="table__body"
          >
            <tr
              class="table__row"
              data-testid="instrument-table-row"
            >
              <td
                class="table__cell "
              >
                OPN2007T
              </td>
              <td
                class="table__cell "
              >
                July 2020
              </td>
              <td
                class="table__cell "
              >
                <a
                  aria-label="Launch interview for instrument OPN2007T July 2020"
                  class="external-link"
                  href="https://external-web-url/OPN2007T?LayoutSet=CATI-Interviewer_Large"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Interview
                  <svg
                    class="svg-icon"
                    viewBox="0 0 12 12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13.5,9H13a.5.5,0,0,0-.5.5v3h-9v-9h3A.5.5,0,0,0,7,3V2.5A.5.5,0,0,0,6.5,2h-4a.5.5,0,0,0-.5.5v11a.5.5,0,0,0,.5.5h11a.5.5,0,0,0,.5-.5v-4A.5.5,0,0,0,13.5,9Z"
                      transform="translate(-2 -1.99)"
                    />
                    <path
                      d="M8.83,7.88a.51.51,0,0,0,.71,0l2.31-2.32,1.28,1.28A.51.51,0,0,0,14,6.49v-4a.52.52,0,0,0-.5-.5h-4A.51.51,0,0,0,9,2.52a.58.58,0,0,0,.14.33l1.28,1.28L8.12,6.46a.51.51,0,0,0,0,.71Z"
                      transform="translate(-2 -1.99)"
                    />
                  </svg>
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </main>
    </div>
    <footer
      class="footer"
    >
      <div
        class="footer--body"
        data-analytics="footer"
      >
        <div
          class="container "
        >
          <div
            class="grid"
          >
            <div
              class="u-mb-m"
            >
              <dl
                aria-label="Blaise Server Manager link to telephone server park"
                class="metadata metadata__list grid grid--gutterless u-cf u-mb-l"
                title="Blaise Server Manager link to telephone server park"
              >
                <dt
                  class="metadata__term grid__col col-6@m"
                >
                   Telephone server park for Blaise Server Manager:
                </dt>
                <dd
                  class="metadata__value grid__col col-6@m"
                />
              </dl>
              <hr
                class="footer__hr"
              />
            </div>
          </div>
          <div
            class="grid grid--flex grid--between"
          >
            <div
              class="grid__col"
            >
              <div
                class="u-mt-m u-mb-s"
              >
                <img
                  alt="Office for National Statistics"
                  src="https://ons-design-system.netlify.app/img/ons-logo-black-en.svg"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`React homepage view surveys page matches Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="phase-banner"
      >
        <div
          class="container "
        >
          <div
            class="grid grid--flex grid--gutterless grid--vertical-center grid--no-wrap"
          >
            <div
              class="grid__col col-auto u-flex-no-grow"
            >
              <h3
                class="phase-banner__badge"
              >
                BETA
              </h3>
            </div>
            <div
              class="grid__col col-auto u-flex-shrink"
            >
              <p
                class="phase-banner__desc u-fs-s u-mb-no"
              >
                This is a new service – your 
                <a
                  href="https://ons.service-now.com/"
                >
                  feedback
                </a>
                 will help us improve it.
              </p>
            </div>
          </div>
        </div>
      </div>
      <header
        class="header header--internal"
      >
        <div
          class="header__top"
          role="banner"
        >
          <div
            class="container"
          >
            <div
              class="header__grid-top grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap "
            >
              <div
                class="grid__col col-auto"
              >
                <a
                  class="header__logo-link"
                  href="/"
                >
                  <picture>
                    <img
                      alt="Office for National Statistics logo"
                      class="header__logo"
                      src="https://ons-design-system.netlify.app/img/ons-logo-neg-en.svg"
                    />
                  </picture>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="header__main"
        >
          <div
            class="container"
          >
            <div
              class="grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap"
            >
              <div
                class="grid__col col-auto u-flex-shrink"
              >
                <div
                  class="header__title"
                >
                  Telephone Operations Blaise Interface
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
      <div
        class="page__container container"
        style="min-height: calc(67vh);"
      >
        <main
          class="page__main"
          id="main-content"
        >
          <h1>
            Interviewing
          </h1>
          <p>
            This page provides information on active questionnaires with corresponding links that redirect to specific areas of CATI dashboard.
          </p>
          <p>
            Please note, the table containing information on active questionnaires may take a few seconds to load.
          </p>
          <p
            class="u-mt-m"
          >
            <a
              class="external-link"
              id="cati-dashboard"
              rel="noopener noreferrer"
              target="_blank"
            >
              Link to CATI dashboard
              <svg
                class="svg-icon"
                viewBox="0 0 12 12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.5,9H13a.5.5,0,0,0-.5.5v3h-9v-9h3A.5.5,0,0,0,7,3V2.5A.5.5,0,0,0,6.5,2h-4a.5.5,0,0,0-.5.5v11a.5.5,0,0,0,.5.5h11a.5.5,0,0,0,.5-.5v-4A.5.5,0,0,0,13.5,9Z"
                  transform="translate(-2 -1.99)"
                />
                <path
                  d="M8.83,7.88a.51.51,0,0,0,.71,0l2.31-2.32,1.28,1.28A.51.51,0,0,0,14,6.49v-4a.52.52,0,0,0-.5-.5h-4A.51.51,0,0,0,9,2.52a.58.58,0,0,0,.14.33l1.28,1.28L8.12,6.46a.51.51,0,0,0,0,.71Z"
                  transform="translate(-2 -1.99)"
                />
              </svg>
            </a>
          </p>
          <h2>
            Surveys
          </h2>
          <table
            class="table "
            id="survey-table"
          >
            <thead
              class="table__head u-mt-m"
            >
              <tr
                class="table__row"
              >
                <th
                  class="table__header "
                  scope="col"
                >
                  <span>
                    Survey three letter acronym
                  </span>
                </th>
                <th
                  class="table__header "
                  scope="col"
                >
                  <span>
                    Link to questionnaires
                  </span>
                </th>
              </tr>
            </thead>
            <tbody
              class="table__body"
            >
              <tr
                class="table__row"
                data-testid="survey-table-row"
              >
                <td
                  class="table__cell "
                >
                  OPN
                </td>
                <td
                  class="table__cell "
                >
                  <a
                    href="/survey/OPN"
                  >
                    View active questionnaires
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
        </main>
      </div>
      <footer
        class="footer"
      >
        <div
          class="footer--body"
          data-analytics="footer"
        >
          <div
            class="container "
          >
            <div
              class="grid"
            >
              <div
                class="u-mb-m"
              >
                <dl
                  aria-label="Blaise Server Manager link to telephone server park"
                  class="metadata metadata__list grid grid--gutterless u-cf u-mb-l"
                  title="Blaise Server Manager link to telephone server park"
                >
                  <dt
                    class="metadata__term grid__col col-6@m"
                  >
                     Telephone server park for Blaise Server Manager:
                  </dt>
                  <dd
                    class="metadata__value grid__col col-6@m"
                  />
                </dl>
                <hr
                  class="footer__hr"
                />
              </div>
            </div>
            <div
              class="grid grid--flex grid--between"
            >
              <div
                class="grid__col"
              >
                <div
                  class="u-mt-m u-mb-s"
                >
                  <img
                    alt="Office for National Statistics"
                    src="https://ons-design-system.netlify.app/img/ons-logo-black-en.svg"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <div
      class="phase-banner"
    >
      <div
        class="container "
      >
        <div
          class="grid grid--flex grid--gutterless grid--vertical-center grid--no-wrap"
        >
          <div
            class="grid__col col-auto u-flex-no-grow"
          >
            <h3
              class="phase-banner__badge"
            >
              BETA
            </h3>
          </div>
          <div
            class="grid__col col-auto u-flex-shrink"
          >
            <p
              class="phase-banner__desc u-fs-s u-mb-no"
            >
              This is a new service – your 
              <a
                href="https://ons.service-now.com/"
              >
                feedback
              </a>
               will help us improve it.
            </p>
          </div>
        </div>
      </div>
    </div>
    <header
      class="header header--internal"
    >
      <div
        class="header__top"
        role="banner"
      >
        <div
          class="container"
        >
          <div
            class="header__grid-top grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap "
          >
            <div
              class="grid__col col-auto"
            >
              <a
                class="header__logo-link"
                href="/"
              >
                <picture>
                  <img
                    alt="Office for National Statistics logo"
                    class="header__logo"
                    src="https://ons-design-system.netlify.app/img/ons-logo-neg-en.svg"
                  />
                </picture>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="header__main"
      >
        <div
          class="container"
        >
          <div
            class="grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap"
          >
            <div
              class="grid__col col-auto u-flex-shrink"
            >
              <div
                class="header__title"
              >
                Telephone Operations Blaise Interface
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
    <div
      class="page__container container"
      style="min-height: calc(67vh);"
    >
      <main
        class="page__main"
        id="main-content"
      >
        <h1>
          Interviewing
        </h1>
        <p>
          This page provides information on active questionnaires with corresponding links that redirect to specific areas of CATI dashboard.
        </p>
        <p>
          Please note, the table containing information on active questionnaires may take a few seconds to load.
        </p>
        <p
          class="u-mt-m"
        >
          <a
            class="external-link"
            id="cati-dashboard"
            rel="noopener noreferrer"
            target="_blank"
          >
            Link to CATI dashboard
            <svg
              class="svg-icon"
              viewBox="0 0 12 12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.5,9H13a.5.5,0,0,0-.5.5v3h-9v-9h3A.5.5,0,0,0,7,3V2.5A.5.5,0,0,0,6.5,2h-4a.5.5,0,0,0-.5.5v11a.5.5,0,0,0,.5.5h11a.5.5,0,0,0,.5-.5v-4A.5.5,0,0,0,13.5,9Z"
                transform="translate(-2 -1.99)"
              />
              <path
                d="M8.83,7.88a.51.51,0,0,0,.71,0l2.31-2.32,1.28,1.28A.51.51,0,0,0,14,6.49v-4a.52.52,0,0,0-.5-.5h-4A.51.51,0,0,0,9,2.52a.58.58,0,0,0,.14.33l1.28,1.28L8.12,6.46a.51.51,0,0,0,0,.71Z"
                transform="translate(-2 -1.99)"
              />
            </svg>
          </a>
        </p>
        <h2>
          Surveys
        </h2>
        <table
          class="table "
          id="survey-table"
        >
          <thead
            class="table__head u-mt-m"
          >
            <tr
              class="table__row"
            >
              <th
                class="table__header "
                scope="col"
              >
                <span>
                  Survey three letter acronym
                </span>
              </th>
              <th
                class="table__header "
                scope="col"
              >
                <span>
                  Link to questionnaires
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="table__body"
          >
            <tr
              class="table__row"
              data-testid="survey-table-row"
            >
              <td
                class="table__cell "
              >
                OPN
              </td>
              <td
                class="table__cell "
              >
                <a
                  href="/survey/OPN"
                >
                  View active questionnaires
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </main>
    </div>
    <footer
      class="footer"
    >
      <div
        class="footer--body"
        data-analytics="footer"
      >
        <div
          class="container "
        >
          <div
            class="grid"
          >
            <div
              class="u-mb-m"
            >
              <dl
                aria-label="Blaise Server Manager link to telephone server park"
                class="metadata metadata__list grid grid--gutterless u-cf u-mb-l"
                title="Blaise Server Manager link to telephone server park"
              >
                <dt
                  class="metadata__term grid__col col-6@m"
                >
                   Telephone server park for Blaise Server Manager:
                </dt>
                <dd
                  class="metadata__value grid__col col-6@m"
                />
              </dl>
              <hr
                class="footer__hr"
              />
            </div>
          </div>
          <div
            class="grid grid--flex grid--between"
          >
            <div
              class="grid__col"
            >
              <div
                class="u-mt-m u-mb-s"
              >
                <img
                  alt="Office for National Statistics"
                  src="https://ons-design-system.netlify.app/img/ons-logo-black-en.svg"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
